pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -157,6 +159,25 @@ func New(args runtime.Object, handle frameworktypes.Handle) (frameworktypes.Plug
 		})
 	}
 
+	// check pod by namespace label filter
+	if defaultEvictorArgs.NamespaceLabelSelector != nil {
```

Comment:
Hello, I think you are right. But i have one question. In front of my code there is a constraint for podLabelSelector. And this selector theoretically does not allow to compute all pods, because we can label pods and remove them from the computation using pod label selector.

Icluding/excluding namespaces by name list is not acceptable for us, because in this case we cannot work with dynamically created namespaces (e.g. for dev environments). Using a label selector is our preferred method.

Reply:
Is this part of https://github.com/kubernetes-sigs/descheduler/issues/1499?

Reply:
> But i have one question.

Which question is it? I see only statements.

Reply:
Yes, it's a part of #1499. The question is, if you can select pods using a label selector, how will the computational resources of all pods work? And why selecting pods by namespace label selector differs from pod label selector in that case ?

Reply:
Changing the code isn't a problem, but I'm trying to figure out how to do it right.

Reply:
> how will the computational resources of all pods work?

I did not get this part. Can you more elaborate on this? Are you asking how each plugin gets a list of all pods in the cheapest way possible? Not sure if this is related to your question but the default evictor's label selector is used for rejecting pods that are not expected to be evicted. So there's no direct relation between the (pod) label selector and listing all the pods.

Reply:
Thank you for your reply. When reading the code I got the same results. 
The Low node utilization plugin lists all pods for a node and calls the pod eviction filter function. 
I want to restrict evicted pods by namespace. For my use case, restricting namespaces by include/exclude is not suitable because namespaces are created dynamically. So I added a namespace label selector. This selector restricts the list of evicted namespaces in the same way as the pod label selector.
 
Do you think this is the wrong way?

Reply:
So the namespace matching (either through included/excluded list or a label selector) can be done at two stages. The first when a plugin constructs a list of pods it wants to process (the more pods are excluded at the beginning the easier/faster a plugin can operate). And when an evictor (currently we have only one) checks which pods are allowed to be evicted. Some plugins like LowNodeUtilization or HighNodeUtilization can not exclude pods at the beginning since they first need to compute the overall resource utilization. So only the pod matching during the eviction step makes sense.

Back to your question. Yes, this is the right way for LowNodeUtilization. The namespace label selector needs to be implemented through the `PreEvictionFilter` extension point. Every plugin can use the new namespace label selector this way. In addition, there are plugins like PodLifeTime that can perform the namespace matching sooner. Which is what is preferable. However, when done sooner each plugin take responsibility for making sure the matching is done in a way that does not break the plugin's intention.

Reply:
ok, thanks, i'll move the label selector code to the `PreEvictionFilter`.

Reply:
Code moved to the `preEvictionFilter`. Please check it again.

Reply:
@ingvagabund hello. Can you help me ? Now namespaces list by label selector calculated for every pod which is passed to the PreEvictionFilter. I think we can calculate namespaces name list and pass it to the PreEvictionFilter to reduce cpu usage (i think we can calculate it on every Descheduler Loop). Please advise where this can be done.

Reply:
You could create a new namespace indexer storing namespaces based on a label selector. Which will give you all the namespaces at any given point for free.

Reply:
Hello, i rebased to current main and add indexer. Can you please check implementation ? If all ok, i'll add tests.

Reply:
@ingvagabund hi, please check implementation if you have time.